nienaruszalnosc-danych.rst

```diff
@@ -139,6 +138,19 @@ System KSI pozwala na udowodnienie, że określony stan bazy danych istniał w k
 2. Odwoływanie się do łańcucha sum kontrolnych w podpisach elektronicznych. Dzięki temu podpis uczciwego człowieka składany na dokumencie w danej sprawie poświadczałby również historię tej sprawy i uniemożliwiał jej zmianę do tego punktu.
 3. Publikację cząstkowych sum kontrolnych w inny sposób (np. publikacja sumy konrolnej sprawy w powiadomieniach emailowych wysyłanych obywatelowi).
 
+Długoterminowa archiwizacja podpisów
+------------------------------------
+
+Tradycyjna metoda konserwacji podpisu polega na opatrywaniu go pieczęcią elektroniczną poświadczającą, że podpis został utworzony przed określonym punktem czasowym (w którym tylko właściwa osoba dysponowała danymi pozwalającymi na złożenie tego podpisu). Pieczęć korzysta przy tym z tej samej technologii, co sam podpis. Zwiększenie mocy obliczeniowej komputerów niesie ze sobą ryzyko złamania zarówno jednego, jak i drugiego. Przyjmuje się, że wejście w życie komputerów kwantowych będzie oznaczało koniec współcześnie popularnych metod podpisywania (por. dokument “`CNSA Suite and Quantum Computing FAQ`_”). Za pomocą komputerów kwantowych będzie można, mając dostęp do publicznych certyfikatów, podrobić zarówno podpis pod dokumentem, jak i pieczęć znakującą ten podpis czasem.
+
+Konserwacja podpisów w oparciu o Keyless Signatures’ Infrastructure bazuje tymczasem nie na tej samej technologii, co podpis (kryptografia klucza publicznego), tylko na funkcjach skrótu, które są uznawane za odporne na zagrożenia związane z informatyką kwantową. Co za tym idzie, o ile ktoś byłby w stanie w przyszłości podrobić podpis stosowany współcześnie, to nie będzie w stanie udowodnić, że podpis ten został współcześnie stworzony, co jest kluczowe przy jego weryfikacji (certyfikaty wykorzystywane do tworzenia podpisów mają swoje okresy ważności).
```

Comment:
Generalnie słuszna uwaga. Zmieniam „nie będzie w stanie udowodnić” na „niekoniecznie będzie w stanie udowodnić”. Tym niemniej nie wiem, czy są obecnie znane konkretne zagrożenia technologiczne dla funkcyj skrótu. Wikipedia pisze następująco w artykule „[Linked timestamping](https://en.wikipedia.org/wiki/Linked_timestamping)”:

> The common technology[2] for guaranteeing long-term attestation value of the issued time-stamps (and digitally signed data[3]) is periodic over-time-stamping of the time-stamp token. Because of missing key-related risks and of the plausible safety margin of the reasonably chosen hash function this over-time-stamping period of hash-linked token could be an order of magnitude longer than of public-key signed token.

[Podobno wejście w życie komputerów kwantowych będzie oznaczało dwukrotne zmniejszenie bezpieczeństwa funkcyj haszujących.](https://bitcointalk.org/index.php?topic=133425.msg1433496#msg1433496). W przypadku SHA-256 będzie to jednak oznaczać bezpieczeństwo na poziomie 128 bitów, czyli potrzebne będzie wykonanie średnio 2<sup>128</sup> haszowań — wystarczająco wiele, aby nadal czuć się bezpiecznym (nie wspominając o możliwości wykorzystania np. funkcji SHA-512). Najnowszy blok z sieci Bitcoina ma obecnie hasz `00000000000000000029416d37d4fb3e4e48fb183fb3e05af1b337f9f7c10acb`, co też daje pewne pojęcie o współczesnej mocy obliczeniowej komputerów (i o tym, jak wielki postęp musiałby się dokonać, aby łamać hasze metodą brute force).